### PR TITLE
staticpod: move revision status configmap update to prune controller

### PR DESCRIPTION
**Pull request by sttts**
When the startup-monitor marks revision as good, the installer controller is not the right place to update the revision status configmap. It fits better into the prune controller that also consumes it.

### CODE REVIEW

**Comment by sttts** [comment]
Rebased.

File: pkg/operator/staticpod/controller/installer/installer_controller.go [hunk 1]
```diff
@@ -654,18 +612,7 @@ func (c *InstallerController) newNodeStateForInstallInProgress(ctx context.Conte
...
-			return nil, false, "", err
-		}
-
-		if currentRevision != strconv.Itoa(int(currNodeState.TargetRevision)) {
-			// new updated pod to be launched
-			if len(currentRevision) == 0 {
-				return ret, false, fmt.Sprintf("waiting for static pod of revision %d", currNodeState.TargetRevision), nil
-			}
+			// =
 			return ret, false, fmt.Sprintf("waiting for static pod of revision %d, found %s", currNodeState.TargetRevision, currentRevision), nil
```
[the remark this answers is not in the document]
Comment (sttts) [reply pkg/operator/staticpod/controller/installer/installer_controller.go:616]: I think this shouldn't be here

[the remark this answers is not in the document]
Comment (sttts) [reply pkg/operator/staticpod/controller/installer/installer_controller.go:616]: I think it was when the cat 🐱  entered the keyboard, see the `// =` above :D

**Review by sanchezl — APPROVED** [review]
/lgtm

**Pull request merged**